core/Fetch.cpp

```diff
@@ -49,52 +70,156 @@ namespace olympia
                                                          workload->getValueAsString(),
                                                          skip_nonuser_mode_);
 
-        fetch_inst_event_->schedule(1);
+        ev_fetch_insts->schedule(1);
     }
 
+
     void Fetch::fetchInstruction_()
     {
-        const uint32_t upper = std::min(credits_inst_queue_, num_insts_to_fetch_);
+        // Prefill the ibuf with some instructions read from the tracefile
+        // keeping enough capacity to group them into cache block accesses.
```

Comment:
Is there a specific reason why we need to prefill instructions from the trace? Can we not read the trace after we get the response back from Icache? When we have decoupled frontend, isn't that what we would expect to do?